pkg/ai/types_ai.go

```diff
@@ -154,3 +118,266 @@ type AiNotificationsDestinationFilter struct {
+func (x AiNotificationsOAuth2Auth) GetRefreshInterval() int {
+	return x.RefreshInterval
+}
+
+// GetRefreshable returns a pointer to the value of Refreshable from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetRefreshable() bool {
+	return x.Refreshable
+}
+
+// GetScope returns a pointer to the value of Scope from AiNotificationsOAuth2Auth
+func (x AiNotificationsOAuth2Auth) GetScope() string {
+	return x.Scope
+}
+
+// AiNotificationsAuth - Authentication interface
+type AiNotificationsAuth struct {
+	// OAuth2 access token url
+	AccessTokenURL string `json:"accessTokenUrl,omitempty"`
+	// Authentication Type - Token, Oauth2, or Basic
+	AuthType AiNotificationsAuthType `json:"authType,omitempty"`
+	// OAuth2 authorization url
+	AuthorizationURL string `json:"authorizationUrl,omitempty"`
+	// OAuth2 clientId
+	ClientId string `json:"clientId,omitempty"`
+	// Token prefix
+	Prefix string `json:"prefix,omitempty"`
+	// Interval of how often should the OAuth2 access token be refreshed
+	RefreshInterval int `json:"refreshInterval,omitempty"`
+	// Is the OAuth2 access token refreshable
+	Refreshable bool `json:"refreshable,omitempty"`
+	// OAuth2 token's scope
+	Scope string `json:"scope,omitempty"`
+	// Basic auth password
+	Password SecureValue `json:"password,omitempty"`
+	// Basic auth user
+	User string `json:"user,omitempty"`
+}
+
+func (x *AiNotificationsAuth) ImplementsAiNotificationsAuth() {}
+
+// AiNotificationsAuth - Authentication interface
+type AiNotificationsAuthInterface interface {
+	ImplementsAiNotificationsAuth()
+	GetAccessTokenURL() string
+	GetAuthType() AiNotificationsAuthType
+	GetAuthorizationURL() string
+	GetClientId() string
+	GetPrefix() string
+	GetRefreshInterval() int
+	GetRefreshable() bool
+	GetScope() string
+}
+
+// UnmarshalAiNotificationsAuthInterface unmarshals the interface into the correct type
+// based on __typename provided by GraphQL
+func UnmarshalAiNotificationsAuthInterface(b []byte) (*AiNotificationsAuthInterface, error) {
+	var err error
+
+	var rawMessageAiNotificationsAuth map[string]*json.RawMessage
+	err = json.Unmarshal(b, &rawMessageAiNotificationsAuth)
+	if err != nil {
+		return nil, err
+	}
+
+	// Nothing to unmarshal
+	if len(rawMessageAiNotificationsAuth) < 1 {
+		return nil, nil
+	}
+
+	var typeName string
+
+	if rawTypeName, ok := rawMessageAiNotificationsAuth["__typename"]; ok {
+		err = json.Unmarshal(*rawTypeName, &typeName)
+		if err != nil {
+			return nil, err
+		}
+
+		switch typeName {
+		case "AiNotificationsBasicAuth":
+			var interfaceType AiNotificationsBasicAuth
+			err = json.Unmarshal(b, &interfaceType)
+			if err != nil {
+				return nil, err
+			}
+
+			var xxx AiNotificationsAuthInterface = &interfaceType
+
+			return &xxx, nil
+		case "AiNotificationsTokenAuth":
+			var interfaceType AiNotificationsTokenAuth
+			err = json.Unmarshal(b, &interfaceType)
+			if err != nil {
+				return nil, err
+			}
+
```

Comment:
we'll need to add the "AiNotificationsOAuth2Auth" here

Reply:
Agreed.

Reply:
Updated